packages/app/src/app/pages/Dashboard/Content/routes/Settings/UserSettings/WorkspaceSettings.tsx

```diff
@@ -25,6 +27,42 @@ export const WorkspaceSettings = () => {
   const { user, activeTeam, activeTeamInfo } = useAppState();
   const actions = useActions();
 
+  const [editing, setEditing] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [file, setFile] = useState<{ name: string; url: string } | null>(null);
+
+  const getFile = async avatar => {
+    const url = await new Promise(resolve => {
+      const reader = new FileReader();
+      reader.onload = e => {
+        resolve(e.target.result);
+      };
+      reader.readAsDataURL(avatar);
```

Comment:
**suggestion:** It may be a good idea to add a `reader.onerror` handler to reject the promise, otherwise the `await` may never end. 😅

Reply:
Sure I can add that - must admit this code was just lifted from the other workspace settings so I did not examine it too closely 🙈